src/libraries/System.Security.Cryptography.Encoding/tests/Cbor/CborReader.cs

```diff
@@ -2,35 +2,105 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+#nullable enable
+using System.Collections.Generic;
+
 namespace System.Security.Cryptography.Encoding.Tests.Cbor
 {
-    internal ref partial struct CborValueReader
+    internal enum CborReaderState
+    {
+        UnsignedInteger = 0,
+        NegativeInteger,
+        ByteString,
+        TextString,
+        StartArray,
+        StartMap,
+        EndArray,
+        EndMap,
+        Tag,
+        Special,
+        Finished,
+        EOF
```

Comment:
Comma after the last item, please :).

Since there's not necessarily a file, perhaps `NoMoreData` or `OutOfData` is a better name?

Reply:
Also, the NoData state is probably the best zero-value.  (So a struct that gets this as a field ends up with NoData if the value is unassigned, instead of having an invalid "it's an unsigned integer, and I have no buffer")

Reply:
> Since there's not necessarily a file, perhaps NoMoreData or OutOfData is a better name?

EOF can be used to refer to data streams in general. I'd favor it since its meaning is pretty standard and unambiguous.

Reply:
All of the "EOF" names (except one in ML.Net that has a lot of other abbreviations) in .NET look to have been added circa .NET Framework 1.0.  There are a lot more things that start with "EndOf" than there are abbreviations EOF.

I'm 98% sure that "EOF" will not survive API review.  The only abbreviations that are allowed are when people know the abbreviation and are less certain what's happening when it's expanded (e.g. more people know "HTML" than "Hyper-Text Markup Language", "JSON" v. "JavaScript Object Notation", "NASA" v. "National Aeronautics and Space Administration").

Reply:
How about `EndOfStream` or `EndOfBuffer`?